module_test.go

```diff
@@ -241,6 +244,85 @@ func TestModuleSuccess(t *testing.T) {
 		)
 		require.NoError(t, app.Err())
 	})
+
+	t.Run("custom logger for module", func(t *testing.T) {
```

Comment:
Let's add a test case that Decorates the logger supplied inside a Module, and make sure that the decorator only gets applied to the Module's logger and not the root-level logger.

Reply:
Added a test case using Replace. Also added another test case for nest Module using parent Module's logger.